src/main.cpp

```diff
@@ -3105,6 +3114,13 @@ bool ContextualCheckBlock(const CBlock& block, CValidationState& state, CBlockIn
         }
     }
 
+    // Check that all transactions are unexpired
+    BOOST_FOREACH(const CTransaction& tx, block.vtx) {
+        if (!tx.IsCoinBase() && IsExpiredTx(tx, nHeight)) {
```

Comment:
If `IsExpired()` contains the coinbase check, no need to include it here. If we decide it won't, then we'd also need to decide here (and correspondingly handle the fact that coinbase can expire in `getblocktemplate`).

Reply:
Are there any potential negative consequences if coinbase txs are excluded from tx expiry checks? If not, then I'd say we should keep that check in IsExpired() for clarity. 

(I put this redundant check here during my debugging process, and forgot to remove.)

Reply:
None that I can think of...